packages/docusaurus/lib/server/load/routes.js

```diff
@@ -5,98 +5,159 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-const {generateChunkName} = require('@docusaurus/utils');
+const {genChunkName, docuHash} = require('@docusaurus/utils');
 const {stringify} = require('querystring');
 
 async function loadRoutes(pluginsRouteConfigs) {
-  const imports = [
+  const routesImports = [
     `import React from 'react';`,
     `import Loadable from 'react-loadable';`,
     `import Loading from '@theme/Loading';`,
     `import NotFound from '@theme/NotFound';`,
   ];
-
+  // Routes paths. Example: ['/', '/docs', '/blog/2017/09/03/test']
   const routesPaths = [];
-  const addRoutesPath = permalink => {
-    if (permalink && !/:|\*/.test(permalink)) {
-      routesPaths.push(permalink);
+  const addRoutesPath = routePath => {
+    routesPaths.push(routePath);
+  };
+  // Mapping of routePath -> metadataPath. Example: '/blog' -> '@generated/metadata/blog-c06.json'
+  const routesMetadataPath = {};
+  const addRoutesMetadataPath = routePath => {
+    const fileName = `${docuHash(routePath)}.json`;
+    routesMetadataPath[routePath] = `@generated/metadata/${fileName}`;
+  };
+  // Mapping of routePath -> metadata. Example: '/blog' -> { isBlogPage: true, permalink: '/blog' }
+  const routesMetadata = {};
```

Comment:
`routesMetadata` and `routesAsyncModules` have the same key and it would be clearer to put them in the same file as:

```json
{
  "/route": {
     "metadata": { ... },
     "asyncModules": [ ... ]
  }
  ...
}
```